source/Missions/BoardGameLogic.cs

```diff
@@ -83,8 +85,26 @@ public void StartGame(bool startFirst, Agent opposingAgent)
             m_BoardGameLogic.SetBoardGame(m_BoardGameType);
             m_BoardGameLogic.SetStartingPlayer(startFirst);
             m_BoardGameLogic.StartBoardGame();
+
+            //Opponent crash detection
+            while(IsPlayingOtherPlayer)
```

Comment:
This will sit in a loop forever, even after the player has won? (`IsPlayingOtherPlayer` isn't being set to false?)

Reply:
It is being set to false, do I know how? no clue.

Reply:
Or actually, maybe not. I don't remember how the BoardGameLogic works entirely

Reply:
If there's a thread still churning round in that loop, no, it'll effectively memory leak (admittedly a small leak), though worse I guess, it'd hold a thread and the .net runtime only allocates a limited number of those to the app domain

Reply:
Actually, looking at it again... I think it results in `Dispose` being called multiple times (or infinite times, not certain). Because you're inside an async void the method is not awaited anywhere, so the method probably blows up, but again, because it's an async void, the exception has nowhere to propagate to. I can pull your branch down to debug it but I suspect _something_ will be doing ugly things right now.

Reply:
Oh, the `Dispose` method is "safe" to repeat I think; nothing in there will blow up if you call it twice as best I can tell? So yeh, you probably just leave a thread sat looping around inside that `while(IsPlayingOtherPlayer)` loop.

Reply:
This can be turned into an event after #465 is merged by subscribing to the PeerDisconnected event

Reply:
Yep, will update after #465

Reply:
Resolving as outdated